x-pack/plugins/apm/server/routes/settings/agent_configuration.ts

```diff
@@ -136,15 +136,19 @@ export const createOrUpdateAgentConfigurationRoute = createRoute(() => ({
   },
 }));
 
+const searchParamsRt = t.intersection([
+  t.type({ service: serviceRt }),
+  t.partial({ etag: t.string, mark_as_applied_by_agent: t.boolean }),
+]);
+
+export type AgentConfigSearchParams = t.TypeOf<typeof searchParamsRt>;
```

Comment:
@dgieselaar would it be better to have a helper that infers the params from a route? Like:
```
type AgentConfigSearchParams = RouteParams<typeof agentConfigurationSearchRoute>
```

Reply:
Sure, you can possibly extract such a helper from the typings we have today. I'm thinking about making `params` a `io-ts` type as well (rather than a record of `io-ts` types), and then it will be even easier.